src/content/blog/ddev-name-resolution-wildcards.md

```diff
@@ -0,0 +1,95 @@
+---
+title: "DDEV Name Resolution, Wildcards, Working Offline"
+pubDate: 2024-03-27
+#modifiedDate: 2024-03-03
+summary: How hostname name resolution works in a DDEV environment, including how wildcards work
+author: Randy Fay
+featureImage:
+  src: /img/blog/2024/03/2024-ddev.png
+  alt: Hostname resolution with DDEV
+  credit: 'Ideogram.ai: the words "2024" and "DDEV" next to each other'
+categories:
+  - Guides
+---
+
+A very cool thing about DDEV is that you don’t need to think about IP addresses or HTTPS or name resolution most of the time. Some of you will never need to. And you don't have to understand teh rest of this article to use DDEV.
+
+This article attempts to unwind what DDEV does for you, and what to do when things require a bit more effort.
+
+## Name resolution for browsers and URLs
+
+The first thing to understand is the structure of a URL used by your browser. A URL like `https://www.google.com/`/search?q=nothing` is structured with 3 main parts, the protocol (usually `https`), the hostname (like `www.google.com`), and the URI, (like `/search?q=nothing`). Every URL that your browser can visit has at least the protocol and the hostname. Sometimes the protocol is hidden implying the default `https`. The URI may be the default "front page".
+
+When your browser gets a URL, no matter how it gets it, and you want to display the page at that URL, the browser has to parse the URL, then turn the hostname into an IP address so it can find its way to it on the internet, then do the proper HTTP request to the internet site, including the URI. The key thing for us right now is that it has to look up that hostname.
+
+Normally with a DDEV local development project, we'll see URLs like `https://something.ddev.site`. It's all done like that so that your local project will work with real URLs and real HTTPS, and essentially behave like a real site. In the past, local development environments often used URLs like `http://127.0.0.1:2302/somesite, which isn't easy to look at and definitely doesn't behave the same in the browser as a "real" URL.
```

Comment:
```suggestion
With a local development project in DDEV, we'll usually see URLs like `https://something.ddev.site`. Everything is handled like that so that your local project will work with real URLs and real HTTPS, and essentially behave like a real site. In the past, local development environments often used URLs like `http://127.0.0.1:2302/somesite, which isn't easy to look at and definitely doesn't behave the same in the browser as a "real" URL.
```

Reply:
Maybe remove 'somesite' since it was the port that was different between projects.

Reply:
```suggestion
With a local development project in DDEV, we'll usually see URLs like `https://example.ddev.site`. Everything is handled like that so that your local project will work with real URLs and real HTTPS, and essentially behave like a real site. In the past, local development environments often used URLs like `http://127.0.0.1:2302/example.com, which isn't easy to look at and definitely doesn't behave the same in the browser as a "real" URL.
```